File: backend/app/schemas/pricing.py
from typing import Optional
from uuid import UUID
from pydantic import BaseModel


class PricingRowCreate(BaseModel):
    wbs_id: Optional[UUID] = None
    person_id: Optional[UUID] = None
    hourly_rate: float = 0
    hours_by_phase: dict = {}


class PricingRowUpdate(BaseModel):
    wbs_id: Optional[UUID] = None
    person_id: Optional[UUID] = None
    hourly_rate: Optional[float] = None
    hours_by_phase: Optional[dict] = None


class PricingRowOut(BaseModel):
    id: UUID
    proposal_id: UUID
    wbs_id: Optional[UUID]
    person_id: Optional[UUID]
    # Denormalised from person for display
    person_name: Optional[str]
    person_wsp_role: Optional[str]
    person_team: Optional[str]
    hourly_rate: float
    hours_by_phase: dict
    total_hours: float
    total_cost: float

    model_config = {"from_attributes": True}

File: frontend/src/api/pricing.ts
import api from "./client";

export interface PricingRow {
  id: string;
  proposal_id: string;
  wbs_id: string | null;
  person_id: string | null;
  // Denormalised from person
  person_name: string | null;
  person_wsp_role: string | null;
  person_team: string | null;
  hourly_rate: number;
  hours_by_phase: Record<string, number>;
  total_hours: number;
  total_cost: number;
}

export const pricingApi = {
  list: (proposalId: string) =>
    api.get<PricingRow[]>(`/api/proposals/${proposalId}/pricing/`).then(r => r.data),
  create: (proposalId: string, data: Partial<PricingRow>) =>
    api.post<PricingRow>(`/api/proposals/${proposalId}/pricing/`, data).then(r => r.data),
  update: (proposalId: string, rowId: string, data: Partial<PricingRow>) =>
    api.patch<PricingRow>(`/api/proposals/${proposalId}/pricing/${rowId}`, data).then(r => r.data),
  delete: (proposalId: string, rowId: string) =>
    api.delete(`/api/proposals/${proposalId}/pricing/${rowId}`),
};
